### DISCUSSION TITLE
Would building an API be worth it?

### DISCUSSION THREAD

**Discussion by alexjustesen** [alexjustesen/speedtest-tracker#22] [Polls]
Had a thought the other night it might be useful to expose an API so that data could be fetched, would this be useful for anyone?

**Comment by Dictynna-PHA** [comment]
Absolutely.
I'm not sure how the original handled it but I can grab the following from it with a plugin from [this](https://github.com/benphelps/homepage) project and it gives a useful output for dashboards.

![speedtest](https://user-images.githubusercontent.com/82800964/190866345-bdfaa933-9818-4e21-8296-ca57a78e3b55.png)

**Reply by alexjustesen** [reply]
Is that a build in integration in that project?

**Reply by alexjustesen** [reply]
Got it, nvm https://github.com/benphelps/homepage/wiki/Service-Widgets#speedtest-tracker

**Comment by alexdelprete** [comment]
Data is in a db, and also in influxdb, so potentially 3rd party apps can take it from there, but an API is always the best in order to be able to change the data structure without breaking integrations.

BTW: are you releasing a new alpha? Would like to test the PR I submitted, to see if it solved my problem, I still haven't been able to do a speedtest with this project. :)

**Reply by alexjustesen** [reply]
I was sitting on a release for a couple extra things but just published https://github.com/alexjustesen/speedtest-tracker/discussions/23 so GDPR works for you

**Comment by alexjustesen** [comment]
API is next up on my todo list, I will be providing backwards compatibility for the current API route `api/speedtest/latest` so home lab dashboard like [Homepage](https://github.com/benphelps/homepage) still work. But I do plan on deprecating this in the future.

Any new endpoints will be developed under a version endpoint.

**Reply by cgawthorpe** [reply]
any update on this?
